packages/swarm-components/src/Textarea.jsx

```diff
@@ -13,25 +18,50 @@ type Props = {
     /**
      * Value for textarea.
      */
-    value?: string
+    value?: string,
+    /**
+     * max length of input field
+     */
+    maxLength?: string | number,
 }
 
 const getTextareaStatus = (props: Props): string => {
     return props.disabled ? 'disabled' : (props.error ? 'error' : 'default');
 };
 
 const getCharacterCount = (value: string = '') => value.length;
+type CharProps = {
+    children?: any
+}
+
+const CharCount = (props: CharProps) => {
+    return (
+        <p data-swarm-textarea-char-count className="text--tiny" {...props} />
+    );
+}
 
-const Textarea = (props: Props): React.Element<'textarea'> => {
+const Textarea = (props: Props): React.Element<'div'> => {
 
-    console.log('textarea char count', getCharacterCount(props.value));
+    const { maxLength, autosize } = props;
+
+    const textareaRef = React.useCallback(node => {
+        if (node !== null && props.autosize) {
+            auto(node);
+        }
+    }, [props.value]);
+
+    const remainingCharacters = (parseInt(maxLength, 10) - getCharacterCount(props.value))
```

Comment:
Doesn't seem like `getCharacterCount` is a necessary function. `props.value` is an optionally required prop, so this could just be :

```
const remainingCharacters = props.value ? parseInt(maxLength,10) - props.value.length : null;
```

Reply:
Flow was giving me an error on saying you can't have length of undefined with something similar to `props.value.length` so I pulled it into a function here.